openwisp_controller/config/static/config/js/widget.js

```diff
@@ -1,6 +1,69 @@
 'use strict';
 (function ($) {
     var inFullScreenMode = false,
+        prevDefaultValues = {},
+        defaultValuesUrl = window.location.origin + '/admin/config/device/get-template-default-values/',
+    removeDefaultValues = function(contextValue, defaultValues) {
+        // remove obsulate default values from contextValue
+        Object.keys(prevDefaultValues).forEach(function (key) {
+            if (!defaultValues.hasOwnProperty(key)) {
+                delete contextValue[key];
+            }
+        });
+        return contextValue;
+    },
+    removeDefaultValuesOnSubmit = function(contextValue) {
```

Comment:
I would rename this `removeUnchangedDefaultValues` and add a preceding commit like:

```
// This method is called on the submit event to remove any template default value which was not customized
// which allows to avoid saving redundant data
```